src/Resolver/ValueResolver/ValueResolver.php

```diff
@@ -35,7 +37,11 @@ public function resolve(Expr $expr, ?string $actualFile = null)
             }
 
             if ($expr instanceof Cast) {
-                return $this->resolve($expr->expr, $actualFile);
+                return $this->resolve($expr->expr, $actualFile, $unknownValuePlaceholder);
+            }
+
+            if ($expr instanceof Variable && $unknownValuePlaceholder) {
```

Comment:
We can resolve variable expression in Path resolver and we don’t need pass $unknownValuePlaceholder to ValueResolver

Reply:
How would you do it wirhour copy paste entire evaluator?

Reply:
I have to check it on desktop, CR on mobile really sucks

Reply:
Do not copy entire evaluator, just add this condition to path resolver

```
if ($expr instanceof Variable && $this->resolveAllPossiblePaths) {
$result = '*'
} else {
// call value resolver
}
```

Reply:
But this piece of code is incide callback passed to ConstExprEvaluator.

Reply:
I would need to extract this callback to be able do add piece of code to it

Reply:
and it contains recursive call that would neet to recursively call pathresolver

Reply:
I know, but that callback is there just because ConstExprEvaluator can't resolve Variable Node. We can handle it before calling ValueResolver. Recursive call is only one as I saw, and it is Casting, so it can be here too.

BUT what about this suggestion https://github.com/efabrica-team/phpstan-latte/pull/84#issuecomment-1333722634?

Reply:
MAybe I am missing something. Can you show me how would you rewrite it?